app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java

```diff
@@ -18,32 +19,68 @@ public SignUpViewModel() {
 
     // TODO: Handle exceptions concerning the failure of the "user" database collection
     public Task<Void> createUser(String email, String password, String phoneNumber, String firstName, String lastName) {
+
+        // Check for phone number if it's in database and unique
+        Task<Void> taskIsPhoneNumberInDatabase =
+                repositoryFacade.getUserRepository().isPhoneNumberInDatabase(phoneNumber);
+
         // add user email and password to authentication database
-        Task<AuthResult> taskAuthSignUpResult = repositoryFacade.getAuthRepository().signUpUser(email, password);
+        Task<AuthResult> taskAuthSignUpResult = taskIsPhoneNumberInDatabase.continueWithTask(task -> {
+            if (task.isSuccessful()) {
+                return signUpUserInAuthRepo(email, password);
+            } else {
+                throw task.getException();
+            }
+        });
```

Comment:
this can be condensed to...

```java
Task<AuthResult> taskAuthSignUpResult = taskIsPhoneNumberInDatabase.onSuccessTask(task -> {
                return signUpUserInAuthRepo(email, password);
        });
```

Although this is minor and nothing too important, so you can ignore this if you want.
Applies to all other cases where the `continueWithTask` simply throws the exception of the previous task when `task.isSuccessful()` is false.

Reply:
The reason I'm doing it this way is to throw an exception just in case the task fails.
I don't want anything else to run after that task. 

Does onSuccessTask do that?

Reply:
yep:
https://developers.google.com/android/reference/com/google/android/gms/tasks/Task#onSuccessTask(com.google.android.gms.tasks.SuccessContinuation%3CTResult,%20TContinuationResult%3E)

Reply:
Oh, that's pretty cool.

However, the only problem I have with it is that it doesn't mention it will throw an exception implicitly; it will simply cancel the task without any indication. I believe that for every task, there should be some exception handling.